front/src/applications/stdcm/hooks/useLinkedPathSearch.ts

```diff
@@ -108,20 +114,19 @@ const useLinkedPathSearch = () => {
       const newLinkedPathResults = await Promise.all(
         filteredResults.map(async (result) => {
           const resultSummary = filteredResultsSummaries && filteredResultsSummaries[result.id];
-          const msFromStartTime =
-            resultSummary && resultSummary.status === 'success'
-              ? resultSummary.path_item_times_final.at(-1)!
-              : 0; // TODO Better handling of this case
+          if (!resultSummary || resultSummary.status !== 'success') return undefined;
+          const msFromStartTime = resultSummary.path_item_times_final.at(-1)!;
 
-          const opDetails = await getExtremitiesDetails(result.path);
+          const originDetails = await getExtremityDetails(result.path.at(0)!);
+          const destinationDetails = await getExtremityDetails(result.path.at(-1)!);
           const computedOpSchedules = computeOpSchedules(result.start_time, msFromStartTime);
 
-          if (opDetails === undefined) return undefined;
+          if (!originDetails || !destinationDetails) return undefined;
           return {
             trainName: result.train_name,
-            origin: { ...opDetails.origin, ...computedOpSchedules.origin } as StdcmLinkedPathStep,
+            origin: { ...originDetails, ...computedOpSchedules.origin } as StdcmLinkedPathStep,
```

Comment:
If you have added the cast in getExtremityDetails, you can remove this one
```suggestion
            origin: { ...originDetails, ...computedOpSchedules.origin },
```